src/main/java/org/jabref/gui/actions/Action.java

```diff
@@ -19,4 +20,71 @@ default Optional<KeyBinding> getKeyBinding() {
     default String getDescription() {
         return "";
     }
+
+    class Builder {
+        private final ActionImpl actionImpl;
+
+        public Builder(String text) {
+            this.actionImpl = new ActionImpl();
+            setText(text);
+        }
+
+        public Builder() {
+            this("");
+        }
+
+        public Action setIcon(JabRefIcon icon) {
+            Objects.requireNonNull(icon);
+            actionImpl.icon = icon;
+            return actionImpl;
+        }
+
+        public Action setText(String text) {
+            Objects.requireNonNull(text);
+            actionImpl.text = text;
+            return actionImpl;
+        }
+
+        public Action setKeyBinding(KeyBinding keyBinding) {
+            Objects.requireNonNull(keyBinding);
+            actionImpl.keyBinding = keyBinding;
+            return actionImpl;
+        }
+
+        public Action setDescription(String description) {
+            Objects.requireNonNull(description);
+            actionImpl.description = description;
+            return actionImpl;
+        }
+    }
+
+    class ActionImpl implements Action {
```

Comment:
Why don't you use the JabRefAction together with the ActionFactory ?

Reply:
I'm not sure how that would work. The idea behind the builder and `ActionImpl` is to create instances of `org.jabref.gui.actions.Action` without implementing it; which can be very verbose.

Reply:
Well, you could either use them as anonymous classes or as separate classes. The latter is probably useful if you are reusing them in multiple placess.